src/amo/components/Page/index.js

```diff
@@ -0,0 +1,44 @@
+/* @flow */
+import makeClassName from 'classnames';
+import config from 'config';
+import * as React from 'react';
+import { withRouter } from 'react-router-dom';
+
+import AppBanner from 'amo/components/AppBanner';
+import Header from 'amo/components/Header';
+import type { ReactRouterLocationType } from 'core/types/router';
+
+import './styles.scss';
+
+type Props = {| children: React.Node, isHomePage?: boolean |};
+
+type InternalProps = {|
+  ...Props,
+  _config: typeof config,
+  location: ReactRouterLocationType,
+|};
+
+export const PageBase = ({
+  _config = config,
+  children,
+  isHomePage = false,
+  location,
+}: InternalProps) => {
+  return (
+    <>
+      <Header isHomePage={isHomePage} location={location} />
+      <div className={makeClassName(isHomePage ? 'Page-homepage' : 'Page')}>
```

Comment:
I would always add `Page` as class name and only add `Page-is-homepage` when needed. That way we can have a default style, a class to rely on no matter what happens, and another style that overrides the default one when it is the homepage.